ledger-service/db-backend/src/main/scala/com/digitalasset/http/dbbackend/Queries.scala

```diff
@@ -193,9 +192,40 @@ sealed abstract class Queries(tablePrefix: String, tpIdCacheMaxEntries: Long)(im
     tpid <- sql"""SELECT tpid FROM $templateIdTableName
                   WHERE (package_id = $packageId AND template_module_name = $moduleName
                       AND template_entity_name = $entityName)""".query[SurrogateTpId].unique
-
   } yield tpid
 
+  final def allOffsetsInformation(optOffsetToUpdate: Option[String])(implicit
```

Comment:
I think you could combine this with the `getTemplateInfoBySurrogateTpId` query and do something along the lines of

```
SELECT t.package_id, t.template_module_name, t.template_entity_name, o.party, o.last_offset
FROM $templateIdTableName t JOIN $ledgerOffsetTableName o ON (t.tpid = o.tpid)
WHERE o.last_offset < $last_offset
```
and then group on the first three columns and convert to a
`Map[domain.ContractTypeId.Resolved, Map[domain.Party, domain.Offset]]`